reference-implementation/lib/readable-stream.js

```diff
@@ -1156,12 +1156,22 @@ function SetUpReadableStreamDefaultControllerFromUnderlyingSource(stream, underl
     return InvokeOrNoop(underlyingSource, 'start', [stream._readableStreamController]);
   }
 
-  function pullAlgorithm() {
-    return PromiseInvokeOrNoop(underlyingSource, 'pull', [stream._readableStreamController]);
+  // eslint-disable-next-line func-style
+  let pullAlgorithm = () => Promise.resolve();
+  const pullMethod = GetMethod(underlyingSource, 'pull');
+  if (pullMethod !== undefined) {
+    pullAlgorithm = () => {
+      return PromiseInvoke(pullMethod, underlyingSource, [stream._readableStreamController]);
+    };
   }
 
-  function cancelAlgorithm(reason) {
-    return PromiseInvokeOrNoop(underlyingSource, 'cancel', [reason]);
+  // eslint-disable-next-line func-style
```

Comment:
We should probably just disable this lint in the eslintrc, given that it's hurting us repeatedly in this way. I don't like one-off lint disables like this in general.

Reply:
Done.